sourcegen-model/src/main/java/io/micronaut/sourcegen/model/ExpressionDef.java

```diff
@@ -1205,6 +1222,28 @@ public enum OpType {
         }
     }
 
+    /**
+     * The string concatenation operation.
+     *
+     * @param left     The left expression
+     * @param right    The right expression
+     * @author Andriy Dmytruk
+     * @since 1.7
+     */
+    @Experimental
+    record StringConcatenation(ExpressionDef left, ExpressionDef right) implements ExpressionDef {
```

Comment:
Not sure about this one. We have `JavaIdioms::concatString` maybe we can introduce an expession but can it be a separate PR